### PR TITLE
Fixes #8641 - Update fog to 1.27

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

**Comment by domcleal** [comment]
The AWS support is now split into its own fog-aws gem.  Could you make a new bundler group for it, so we can start relying less on 'fog'?  I think bundler.d (require => 'fog/aws') and config/application.rb are the two places that will need updating.

This will also help fix #2017.

File: bundler.d/fog.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 group :fog do
- gem 'fog', '1.25.0'
- gem 'fog-core', '1.25.0'
+ gem 'fog', '1.27.0'
+ gem 'fog-core', '1.27.3'
  gem 'unf', '>= 0.1.3'
```
Comment (domcleal) [inline bundler.d/fog.rb]: unf has been dropped since https://github.com/fog/fog/commit/8426fc9abf10c0ce2b039d26830e000361dac604 it seems.

**Comment by dLobatog** [comment]
@domcleal Updated

**Comment by domcleal** [comment]
@eLobato I don't see the 'aws' bundler group file?

**Comment by dLobatog** [comment]
@domcleal Sorry!

File: bundler.d/aws.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+group :aws do
+  gem 'fog-aws'
```
Comment (domcleal) [inline bundler.d/aws.rb]: Should we pin it to ~> 0.0 or similar?

Comment (dLobatog) [inline bundler.d/aws.rb]: I'm not sure, it depends on fog-core though so I don't think it'll get any breaking changes if we're already pinning the others. Given it's a really new gem (24 days old) I'd just go without pinning

Comment (domcleal) [inline bundler.d/aws.rb]: I was thinking if fog-aws introduced some breaking change they might bump to 1.x?  Not sure..

Comment (shlomizadok) [inline bundler.d/aws.rb]: Isn't fog-aws added automatically to fog? (https://github.com/fog/fog/blob/master/fog.gemspec#L57)

Comment (domcleal) [inline bundler.d/aws.rb]: Adding its own bundler group means you'll be able to --exclude=fog (and others) for a minimal and faster install.  Our packages are split into one per bundler group, so you'll be able to install foreman-aws (or should that be "ec2"?) also.

File: bundler.d/fog.rb [hunk 3]
```diff
@@ -1,5 +1,4 @@
 group :fog do
- gem 'fog', '1.25.0'
- gem 'fog-core', '1.25.0'
- gem 'unf', '>= 0.1.3'
+ gem 'fog', '1.27.0'
+ gem 'fog-core', '1.27.3'
```
Comment (domcleal) [inline bundler.d/fog.rb]: .4 is out (sorry!)

**Comment by dLobatog** [comment]
@domcleal no problem, updated to 1.27.4

File: config/application.rb [hunk 4]
```diff
@@ -21,7 +21,7 @@
     end
     Bundler.require(*Rails.groups(:assets => %w(development test)))
     if SETTINGS[:unattended]
-      %w[fog libvirt ovirt vmware gce].each do |group|
+      %w[aws fog libvirt ovirt vmware gce].each do |group|
```
Comment (domcleal) [inline config/application.rb]: ec2

**Comment by domcleal** [comment]
I've got the RPM packaging pretty much done, we'll merge this after FOSDEM et al to avoid breaking everything for a few days.

**Comment by dLobatog** [comment]
@domcleal FOSDEM is over :smile:

**Comment by domcleal** [comment]
So it is :)  Thanks @eLobato, merged as 20b093b.

**Pull request closed without merging**